platform/security/common/src/main/java/ddf/security/samlp/impl/SamlValidator.java

```diff
@@ -287,6 +288,24 @@ private Request(Builder builder) {
       logoutRequest = (LogoutRequest) builder.xmlObject;
     }
 
+    /**
+     * A SAML LogoutRequest must include at least one SessionIndex, if the logout is initiated by
+     * Session Participant. However, if the Session Authority initiates the logout, a SessionIndex
+     * is optional. In that case, the absence of a SessionIndex means to logout of all sessions
+     * associated with the Principle. Because the SAMLValidator does not have enough information to
+     * determine who initiated the logout, this method does not throw a ValidationException.
+     * Instead, it logs the condition.
+     *
+     * @throws ValidationException
+     */
+    protected void checkSessionIndexes() throws ValidationException {
```

Comment:
I don't know. There are cases when the lack of a session index could be a legitimate error per the specification. The specification says the logout request MUST have a session index if the logout is initiated by a Session Participant. However, the session index index is OPTIONAL if the logout is initiated by a Session Authority. According to Scott, we only support the former. So, in theory, any logout request without a session index contravenes the specification. However, security isn't really jeopardized if the session index is missing; the IDP logs out the user out of ALL session regardless.

I will remove the `throws ValidationException` and the javadoc comment. Just know that sometime in the future we may to throw that exception.

Reply:
I guess my comment regarding the exception was conditional to making the method package-private. If we keep it `protected`, we should keep the exception in the signature otherwise we'll have to break the API down the road.

I personally would prefer to make it package-private for now until we have a good case for doing otherwise. Just my 2 cents.